src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -450,15 +461,29 @@ class HuiMapCard extends LitElement implements LovelaceCard {
           const opacityStep = 0.8 / (path.length - 2);
           const opacity = 0.2 + markerIndex * opacityStep;
 
+          let marker = Leaflet.circleMarker(path[markerIndex], {
+            radius: 3,
+            color: this._getColor(entityId),
+            opacity,
+            interactive: false,
+          });
+
           // DRAW history path dots
-          mapPaths.push(
-            Leaflet.circleMarker(path[markerIndex], {
-              radius: 3,
-              color: this._getColor(entityId),
-              opacity,
-              interactive: false,
-            })
-          );
+          if (this._config!.show_timestamps) {
+            mapPaths.push(
+              marker
+                .bindTooltip(
+                  format(
```

Comment:
We have existing helpers in src/common/datetime/* that should be reused.

Reply:
The map with date/time additions.
![map](https://user-images.githubusercontent.com/280646/111768837-b9526480-88a8-11eb-9372-4a765fd5cd1b.png)

New fields in map edit
![map_edit](https://user-images.githubusercontent.com/280646/111768834-b8b9ce00-88a8-11eb-8cd1-e60811569044.png)

Reply:
Will try to figure out format_date_time to replace my own variant.

Reply:
Please add the screenshots to your PR description.